jupyter-docker/Dockerfile

```diff
@@ -121,16 +114,17 @@ RUN export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)" \
  && pip install matplotlib-venn \
  && pip install "https://github.com/all-of-us/workbench/archive/pyclient-v0-1-rc3.zip#egg=aou_workbench_client&subdirectory=client/py"
 
-
-ADD spark-defaults.conf.template $SPARK_HOME/conf/
-# Spark defaults conf find and replace [HAILHASH] with the hash
-RUN sed "s|\[HAILHASH\]|$HAILHASH|g" $SPARK_HOME/conf/spark-defaults.conf.template > $SPARK_HOME/conf/spark-defaults.conf
+ADD spark_install_hail.sh $HAIL_HOME/
 ADD jupyter_notebook_config.py $JUPYTER_HOME/
 ADD jupyter_localize_extension.py $JUPYTER_HOME/custom/
-RUN chown -R $USER:users $JUPYTER_HOME
+ADD jupyter_install_notebook_extension.sh $JUPYTER_HOME/
+
+RUN chown -R $USER:users $JUPYTER_HOME \
+  && chmod +x $JUPYTER_HOME/jupyter_install_notebook_extension.sh \
+  && chmod +x $HAIL_HOME/spark_install_hail.sh
 
 USER $USER
 WORKDIR $HOME
 
 EXPOSE $JUPYTER_PORT
-ENTRYPOINT ["pyspark"]
+ENTRYPOINT ["pyspark"]
```

Comment:
Amazingly we can leave the `pyspark` entrypoint.

1. Spark/hadoop is already mounted by the time pyspark runs
2. Hail is _not_ yet installed when pyspark runs, but it is installed by the time the Jupyter _kernel_ starts (which is all we need).

Reply:
Is #2 dependent on some lucky timing magic?

Reply:
It's dependent on how Jupyter/pyspark works -- e.g. it doesn't launch spark until you launch a kernel. I don't think there are race conditions as a kernel can't possibly be started before `init-actions.sh` completes (since the cluster isn't ready). 

Alternatively I did try removing `ENTRYPOINT ["pyspark"]` and doing `docker exec -d pyspark` in the init script. I got some funky errors but I was probably doing something wrong.